src/create_embeddings.py

```diff
@@ -0,0 +1,68 @@
+import pandas as pd
+from core.embeddings.base import Embeddings
+from core.embeddings.bedrock import BedrockEmbeddings
+from utils.logging import logger
+
+
+def _initialize_embedding_df(
+    df: pd.DataFrame, output_csv_filepath: str, target_col: str, embed_col: str
+):
+    try:
+        embedding_df = pd.read_csv(output_csv_filepath)
+    except FileNotFoundError:
+        df = df.dropna(subset=[target_col])
+        group_df = df.groupby(target_col)
+        embedding_df = (
+            group_df[df.columns].apply(lambda x: x.index.tolist()).reset_index(name="index_ids")
+        )
+        embedding_df[embed_col] = None
+    return embedding_df
+
+
+def _find_first_nan_index(df: pd.DataFrame, embed_col: str = "embedding"):
+    nan_index = df[embed_col].isna().idxmax()
+    value = df.loc[nan_index, embed_col]
+
+    if not isinstance(value, list) and pd.isna(value):
+        return nan_index
+    return None
+
+
+def create_embeddings(
+    embed_model: Embeddings,
+    df: pd.DataFrame,
+    output_csv_filepath: str,
+    chunk_size: int = 1000,
+    target_col: str = "label",
+    embed_col: str = "embedding",
+):
+    if target_col not in df.columns:
+        raise ValueError(f"Column with name '{target_col}' should be present in the dataframe.")
+
+    logger.info(f"'{embed_model.model_id}' used for Embedding Text")
+
+    embedding_df = _initialize_embedding_df(df, output_csv_filepath, target_col, embed_col)
+    start_index = _find_first_nan_index(embedding_df)
+
+    if start_index and start_index != 0:
+        logger.info(
+            f"{start_index} rows already embedded. Resume embedding from {start_index + 1}..."
+        )
+
+    if start_index is None:
+        logger.info("All labels already embedded.")
+        return embedding_df
```

Comment:
```suggestion
    if start_index and start_index != 0:
         logger.info(
            f"{start_index} rows already embedded. Resume embedding from {start_index + 1}..."
        )
    elif start_index is None:
        logger.info("All labels already embedded.")
        return embedding_df
```

this way you only check the second condition when needed